includes/payments/actions.php

```diff
@@ -27,19 +27,20 @@
 */
 function edd_complete_purchase( $order_id, $new_status, $old_status ) {
 
+	$completed_statuses = array( 'publish', 'complete', 'completed' );
```

Comment:
That's a great question. We could add a filter and allow recurring to add `edd_subscription`, but I just did some looking and it looks like this function doesn't run currently for renewals (which may be an error).

I suspect this is why renewal payments do not currently have a completed date, which I know is an issue we ran into with Invoices.

So, this won't work with Recurring, but I think it never has, so wondering if changing it would be imprudent. A fair amount of what's done in this function is deprecated/irrelevant in 3.0, or will be--calculating sales/earnings for the customers and downloads, because we're now handling those calculations on status changes.

Reply:
@robincornett That's fair.

I just checked in EDD 3.0 on Sugar Calendar and noticed that After Payment Actions are still processing for renewal payments. And they do have a date completed listed on them.